src/pro/java/hw8/ArrayValueCalculator.java

```diff
@@ -0,0 +1,41 @@
+package pro.java.hw8;
+
+public class ArrayValueCalculator {
+
+    public static void main(String[] args) {
+        String[][] array = {
+                {"1", "2", "3", "4"},
+                {"5", "6", "7", "8"},
+                {"9", "10", "11", "12"},
+                {"13", "14", "15", "16"}
+        };
+
+        try {
+            int calcArray = doCalc(array);
+            System.out.println("Summary result: " + calcArray);
+        } catch (ArraySizeException | ArrayDataException e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    public static int doCalc(String[][] array) throws ArraySizeException, ArrayDataException {
+        if (array.length != 4 || array[0].length != 4) {
+            throw new ArraySizeException("Incorrect array size");
```

Comment:
What I recommend here  - is to overload constructor of ArrayDataException with additional parameter Trowable and create ArrayDataException with message and Exception from catch block do not loose original stacktrace of exception

Reply:
done